src/js/search-bar/search-bar.view.js

```diff
@@ -4,26 +4,65 @@ import {selectDomElement} from "../utils/dom-selector";
 import {SearchBarEvent} from "./search-bar.constants";
 
 const template = `
-    <button id="awesome-button">click</button>
+    <div class="search-bar initial-search-bar" id="search-bar">
+        <input class="search-bar__input" id="search-input" type="search" placeholder="Search...">
+        <button class ="search-bar__icon" id="search-btn">
+            <img src="https://i.ibb.co/d5rcTzD/icons8-magnifying-glass-64.png" alt="search-icon" class ="search-bar__img">
+        </button>
+    </div>
 `
 
 export class SearchBarView extends AbstractView {
 
-    awesomeButton;
+    searchButton;
 
     onViewInit() {
         super.onViewInit();
-        this.selector = '#search-bar';
+        this.selector = '#search-system';
         this.template = template;
     }
 
     afterViewInit() {
         super.afterViewInit();
-        this.awesomeButton = selectDomElement('#awesome-button');
-        this.awesomeButton.onclick = this.sendAwesomeEvent.bind(this);
+        this.searchBar = selectDomElement('#search-bar');
+        this.searchBtn = selectDomElement('#search-btn');
+        this.searchInput = selectDomElement('#search-input');
+
+        this.searchInput.onkeyup = (event) => {
+            if(event.key === 'Enter') {
+                this.sendEvent(SearchBarEvent.CLICK_SEARCH);
+                this.deleteSearchText();
+            }
+        }
+
+        this.searchBar.onclick = () => {
+            if (this.searchBar.classList.contains('initial-search-bar')) {
```

Comment:
you may have a class property isToggled, isActive or isOpened with boolean value and use it in if clause, but not check if DOM element includes the specified class. such approach is cheaper.